Month 7/step 1

MADeit0 opened the pull request.

Commit:
feat: add fonts, create a folder with future components

Commit:
feat: create App-Header component

Commit:
feat: create Burger-Ingredients and Ingredients componets

Commit:
feat: create ingredients-Board component

and set up reading data from the data.js

Commit:
feat: create a class for elements containing ingredients inside

Commit:
feat: fix the classes of the components included in the creation of burgers

Commit:
feat: create Burger-Constructor component

Commit:
feat: adjust styles to use sliders

Commit:
feat: create a slider and add prop types

gennady-bars commented on src/components/App/App.jsx:
```diff
@@ -0,0 +1,21 @@
+  return (
+    <div className={`${appStyle.container} pb-10`}>
+      <AppHeader />
+      <div className={appStyle.section}>
```
`      <div className={appStyle.section}>
`

тут вместо `div` нужно указать `main`

По семантике нужно помещать весь главный контент страницы в тег `main`.

gennady-bars commented on src/components/Burger-Constructor/Burger-Constructor.jsx:
```diff
@@ -0,0 +1,71 @@
+import burgerConstructorsStyle from "./Burger-Constructor.module.css";
+import PropTypes from "prop-types";
+
+const dataPropTypes = PropTypes.shape({
```
тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно

gennady-bars commented on src/components/Ingredients-Board/Ingredients-Board.jsx:
```diff
@@ -0,0 +1,36 @@
+import IngredientsBoardStyle from "./Ingredients-Board.module.css";
+
+const dataPropTypes = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
```
тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно

gennady-bars replied:
и тут не только `_id`. Там куча данных в ингредиенте

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* Верстка отличная


но есть некоторые недочеты: 


* В верхней булке нужно добавлять текст `верх`, а в нижней - `низ`. Посмотрите в фигме, пожалуйста
* Внутри заказа не должно быть **других булок**: есть только верхняя и нижняя одинаковые.
* По семантике нужно помещать весь главный контент страницы в тег `main`.
* тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно

Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 


Удачного рефакторинга кода.

Commit:
bugfix: fix spelling mistakes

add text showing the position of the bun.
include tag main in the layout.
move propTypes to check the array of input data into a separate file.

gennady-bars commented on src/components/Ingridient/Ingredient.jsx:
```diff
@@ -22,10 +21,4 @@ const Ingridient = ({ name, price, image }) => {
   );
 };
 
-Ingridient.propTypes = {
```
`Ingridient` нужно вернуть типизацию. В пропсы приходит `name, price, image`


На самом деле, там приходит весь `ingredientType` тип


и будет вот ак

```js

Ingridient.propTypes = ingredientType.isRequired


```



Но лучше было бы передавать объект ингредиента целиком под названием `ingredient`, 


```js

              <Ingredient ingredient={item} />



```


тогда тут было бы вот так


```js

Ingridient.propTypes = {
  ingredient: ingredientType.isRequired
}
```

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.


p.s. теперь нужно сделать 2й этап с попапами


* `Ingridient` нужно вернуть типизацию. В пропсы приходит `name, price, image`

Commit:
bugfix: return the typing for the Ingredient component
